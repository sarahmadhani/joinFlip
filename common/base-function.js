/* eslint-disable import/prefer-default-export */
import fs from 'fs';

let d;
export const getToken = (namaToken) => {
  const ambilToken = fs.readFileSync('./token.json');
  const objToken = JSON.parse(ambilToken);
  const token = objToken[namaToken];
  return token;
};

export const getTokenMember = (namaToken) => {
  const ambilToken = fs.readFileSync('./tokenMember.json');
  const objToken = JSON.parse(ambilToken);
  const token = objToken[namaToken];
  return token;
};

export function nextMinutes(min) {
  d = new Date();
  d.setMinutes(d.getMinutes() + min);
  return d.toISOString();
}
export function nextHours(hour) {
  d = new Date();
  d.setHours(d.getHours() + hour);
  return d.toISOString();
}
export function pastHours(hour) {
  d = new Date();
  d.setHours(d.getHours() - hour);
  return d.toISOString();
}
export function pastMonth(month) {
  d = new Date();
  d.setMonth(d.getMonth() - month);
  return d.toISOString();
}
export function nextMonth(month) {
  d = new Date();
  d.setMonth(d.getMonth() + month);
  return d.toISOString();
}
export function nextDate(days) {
  d = new Date();
  d.setDate(d.getDate() + days);
  return d.toISOString();
}
export function currentDate() {
  d = new Date();
  return d.toISOString();
}
export function pastDate(days) {
  d = new Date();
  d.setDate(d.getDate() - days);
  return d.toISOString();
}
export function pastYear(year) {
  d = new Date();
  d.setFullYear(d.getFullYear() - year);
  return d.toISOString();
 }
 export function nextYear(year) {
  d = new Date();
  d.setFullYear(d.getFullYear() + year);
  return d.toISOString();
 }