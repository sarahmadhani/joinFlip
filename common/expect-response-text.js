/* eslint-disable radix */
const chai = require('chai');

const expectChai = chai.expect;
chai.use(require('chai-sorted'));

export const output = response => ({
  text: `\n  Request: ${JSON.stringify(response.request, null, 4)} \n  Response: ${JSON.stringify(response.text)}`,
});

export const expectStatus = (response, expectedStatus) => {
  expect((response).status, `Reason: Wrong HTTP Status Code ${output(response).text}`).toEqual(expectedStatus);
};

export const expectBody = (response, resp, expectedBody) => {
  expect(resp, `Reason: Expect and Actual is not Equal ${JSON.stringify(response.text, null, 4)}`).toEqual(expectedBody);
};

export const expectStatusError = (response, resp, expectedStatus) => {
  expect(resp, `Reason: Wrong HTTP Status Code ${JSON.stringify(response, null, 4)}`).toBe(expectedStatus);
};

export const expectErrorStatus = (response, expectedStatus) => {
  const respObj = JSON.parse(response.text);
  if (respObj.errors === undefined) {
    expect(response.status, `Reason: Wrong HTTP Status Code ${output(response).text}`).toEqual(expectedStatus);
  } else {
    expect(respObj.errors[0].httpStatusCode, `Reason: Wrong HTTP Status Code ${output(response).text}`).toEqual(expectedStatus);
  }
};

export const expectEqual = (response, actualData, expectedBody) => {
  expect(actualData, `Reason: Actual Data is not equal with Expected Data ${output(response).text}`).toEqual(expectedBody);
};

export const expectContain = (response, actualData, expectedBody) => {
  expect(actualData, `Reason: Actual Data is not contain with Expected Data ${output(response).text}`).toContain(expectedBody);
};

export const expectToMatchObject = (response, actualData, expectedBody) => {
  expect(actualData, `Reason: Actual Data is not to match obaject with Expected Data ${output(response).text}`).toMatchObject(expectedBody);
};

export const expectSchema = (response, expectedSchema) => {
  expect((response).body, `Reason: Wrong Response Schema ${output(response).text}`).toMatchSchema(expectedSchema);
};

export const expectSchemaSpec = (response, responseSpec, expectedSchema) => {
  expect(responseSpec, `Reason: Wrong Response Schema ${output(response).text}`).toMatchSchema(expectedSchema);
};

export const expectErrRespSchema = (response, expectedSchema) => {
  expect((response).text, `Reason: Wrong Response Schema ${output(response).text}`).toMatchSchema(expectedSchema);
};

export const expectSnapshot = (response) => {
  expect((response).text, `Reason: Response Should be the same with Snapshot ${output(response).text}`).toMatchSnapshot();
};

export const expectSnapEx = (response, snapshotExcept) => {
  expect((response).text, `Reason: Response Should be the same with Snapshot ${output(response).text}`).toMatchSnapshot(snapshotExcept);
};

export const expectSnapshotSpec = (response, actualData) => {
  expect(actualData, `Reason: Response Should be the same with Snapshot ${output(response).text}`).toMatchSnapshot();
};

export const expectSnapExSpec = (response, actualData, snapshotExcept) => {
  expect(actualData, `Reason: Response Should be the same with Snapshot ${output(response).text}`).toMatchSnapshot(snapshotExcept);
};

export const expectDateIsAfter = (response, respDate, expDate) => {
  const expectDate = Date.parse(expDate);
  const responseDate = Date.parse(respDate);
  expect(responseDate, `Reason: Response Should be the same with Snapshot ${output(response).text}`).toBeGreaterThanOrEqual(expectDate);
};

export const expectDateIsBefore = (response, respDate, expDate) => {
  const expectDate = Date.parse(expDate);
  const responseDate = Date.parse(respDate);
  expect(responseDate, `Reason: Response Should be the same with Snapshot ${output(response).text}`).toBeLessThanOrEqual(expectDate);
};

export const expectAscending = (response, actualData) => {
  expectChai(actualData, `Reason: Response must be sorted Ascending ${output(response).text}`).to.be.sorted({ descending: false });
};

export const expectDescending = (response, actualData) => {
  expectChai(actualData, `Reason: Response mush be sorted Descending ${output(response).text}`).to.be.sorted({ descending: true });
};

export const expectLength = (response, actualData, length) => {
  expectChai(actualData, `Reason: Actual Length is not Equal with Expected Length ${output(response).text}`).to.have.lengthOf(length);
};

export const expectLengthWithin = (response, actualData, minLength, maxLength) => {
  expectChai(actualData, `Reason: Actual Length is not Equal between Expected Length ${output(response).text}`).to.have.lengthOf.within(minLength, maxLength);
};

export const expectWithin = (response, actualData, min, max) => {
  expectChai(actualData, `Reason: Actual Number is not Equal between Expected Number ${output(response).text}`).to.be.within(min, max);
};

export const expectGreaterThanOrEqual = (response, actualData, number) => {
  expect(actualData, `Reason: Actual Number is not Greater than or Equal with Expected Number ${output(response).text}`).toBeGreaterThanOrEqual(number);
};

export const expectObjectFillter = (response, actualData, expectData) => {
  expectChai(actualData, `Reason: Actual Data is not Contain or Equal with Expected Data ${output(response).text}`).to.contain.members(expectData);
};


export const expectLessThanOrEqual = (response, actualData, number) => {
  expect(actualData, `Reason: Actual Number is not Less than or Equal with Expected Number ${output(response).text}`).toBeLessThanOrEqual(number);
};

export const expectNotEqual = (response, actualData, expectedBody) => {
  expect(actualData, `Reason: Actual Data should not equal with Expected Data ${output(response).text}`).not.toEqual(expectedBody);
};

export const expectNotContain = (response, actualData, expectedBody) => {
  expect(actualData, `Reason: Actual Data is not contain with Expected Data ${output(response).text}`).not.toContain(expectedBody);
};

export const expectASCBy = (response, ascendingBy) => {
  expectChai((response), `Reason: Response mush be sorted Descending ${output(response).text}`).to.be.sortedBy(`${ascendingBy}`, { ascending: true });
};
export const expectDESCBy = (response, descendingBy) => {
  expectChai((response), `Reason: Response mush be sorted Descending ${output(response).text}`).to.be.sortedBy(`${descendingBy}`, { descending: true });
};

export const expectEmpty = (response, actualData) => {
  expectChai((actualData), `Reason: Response mush be sorted Descending ${output(response).text}`).to.be.empty;
};
