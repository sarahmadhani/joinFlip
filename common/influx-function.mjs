import Influx from 'influx';
import dotenv from 'dotenv';

dotenv.config()

const dbName = 'natacaraATAPI';
const dataType = 'atResult';
const source = 'all-test'

const influx = new Influx.InfluxDB({
  host: process.env.URL_INFLUXDB,
  port: process.env.PORT_INFLUXDB,
  database: dbName,
  schema: [{
    measurement: dataType,
    fields: {
      totalTestSuites: Influx.FieldType.INTEGER,
      totalTestCases: Influx.FieldType.INTEGER,
      testSuitesFailed: Influx.FieldType.INTEGER,
      testSuitesPassed: Influx.FieldType.INTEGER,
      testCasesFailed: Influx.FieldType.INTEGER,
      testCasesPassed: Influx.FieldType.INTEGER,
      testCasesPending: Influx.FieldType.INTEGER,
      testCasesUndefined: Influx.FieldType.INTEGER,
      dataSource: Influx.FieldType.STRING,
    },
    tags: [
      'jobName',
    ],
  }],
});

influx.alterRetentionPolicy('autogen', {
  database: dbName,
  duration: '4w',
  replication: 1,
  default: false,
});

export const writeDataAll = (jsonReport) => {
  const jobName = 'all';
  influx.getDatabaseNames()
    .then((names) => {
      if (!names.includes(dbName)) {
        return influx.createDatabase(dbName);
      }
    })
    .catch((err) => {
      console.log(err);
      console.error('Error created Influx database!');
    })
    .then(() => {
      influx.writePoints([{
        measurement: dataType,
        tags: {
          jobName,
        },
        fields: {
          totalTestSuites: jsonReport.numFailedTestSuites + jsonReport.numPassedTestSuites,
          testSuitesFailed: jsonReport.numFailedTestSuites,
          testSuitesPassed: jsonReport.numPassedTestSuites,
          totalTestCases: jsonReport.numFailedTests + jsonReport.numPassedTests + jsonReport.numPendingTests,
          testCasesFailed: jsonReport.numFailedTests,
          testCasesPassed: jsonReport.numPassedTests,
          testCasesPending: jsonReport.numPendingTests,
          dataSource: source,
        },
      }]);
    });
};

export const writeDataJob = (daftarFolder) => {
  const jobName = daftarFolder.name;
  influx.getDatabaseNames()
    .then((names) => {
      if (!names.includes(dbName)) {
        return influx.createDatabase(dbName);
      }
    })
    .catch((err) => {
      console.log(err);
      console.error('Error created Influx database!');
    })
    .then(() => {
      influx.writePoints([{
        measurement: dataType,
        tags: { jobName },
        fields: {
          totalTestCases: daftarFolder.totalPass + daftarFolder.totalFail + daftarFolder.totalUndefined + daftarFolder.totalPending,
          testCasesFailed: daftarFolder.totalFail,
          testCasesPassed: daftarFolder.totalPass,
          testCasesPending: daftarFolder.totalPending,
          testCasesUndefined: daftarFolder.totalUndefined,
          dataSource: source,
        },
      }]);
    });
};
