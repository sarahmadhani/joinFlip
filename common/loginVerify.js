export const loginVerify = (param) => ({
    mutation: {
        loginVerify: {
            __args: param,
            stage: true,
            user: {
                id: true,
                provider: true,
                username: true,
                email: true,
                phonenumber: true,
                profile: {
                    firstname: true,
                    lastname: true,
                    address: true,
                    phonenumber: true,
                    gender: true,
                    birthdate: true,
                    birthplace: true,
                    hobby: true,
                },
                updatedAt: true,
            },
            authPayload:{
                tokenType: true,
                expiresIn: true,
                scope: true,
                accessToken: true,
                refreshToken: true,
                idToken: true,
            }
        },
    },
});
