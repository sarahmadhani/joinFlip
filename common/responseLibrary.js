export default {
  statusOk: 200,
  statusCreated: 201,
  statusNoContent: 204,
  statusBadRequest: 400,
  statusUnauthorized: 401,
  statusForbidden: 403,
  statusNotFound: 404,
  statusUnprocessableEntity: 422,
  statusInternalServerError: 500,
  statusTooManyRequest: 429,
};

export const responseHandle = (resp) => {
  const newResponse = {
    ...resp,
    body: Buffer.isBuffer(resp.body) ? JSON.parse(resp.body.toString()) : resp.body,
  };

  return newResponse;
};
