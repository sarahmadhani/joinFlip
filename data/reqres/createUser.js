export const data = {
  name: 'morpheus',
  job: 'leader',
};
