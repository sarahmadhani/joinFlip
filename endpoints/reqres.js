import supertest from 'supertest';

const env = require('dotenv').config();

const api = supertest('https://reqres.in');

export const endpointGetUsers = async (param) => api.get(`/api/users?page=${param}`)
  .set('Accept', 'application/json') ;


export const endpointCreateUser = async (body) => api.post('/api/users')
.set('Accept', 'application/json')
.send(body) ;
