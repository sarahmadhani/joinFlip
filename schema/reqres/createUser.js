export default {
  $schema: 'http://json-schema.org/draft-07/schema',
  type: 'object',
  required: ['name', 'job', 'id', 'createdAt'],
  properties: {
    name: {
      type: 'string',
    },
    job: {
      type: 'string',
    },
    id: {
      type: 'string',
    },
    createdAt: {
      type: 'string',
    },
  },
  additionalProperties: false,
};
