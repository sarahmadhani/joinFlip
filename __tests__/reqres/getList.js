import shell from 'shelljs';
import * as er from '../../common/expect-response';
import status from '../../common/responseLibrary';
import * as epEvent from '../../endpoints/reqres';
import * as schema from '../../schema/reqres/getList';

const { matchers } = require('jest-json-schema');
expect.extend(matchers);

let response;

describe('Positive Test - get list', () => {
  test('As user, i can get list users', async () => {
    response = await epEvent.endpointGetUsers(2);
    console.log(JSON.stringify(response.body));
    er.expectStatus(response, status.statusOk);
    er.expectSchema(response, schema.default);
    er.expectSnapshot(response, data.snapshotEx);
  });
});
