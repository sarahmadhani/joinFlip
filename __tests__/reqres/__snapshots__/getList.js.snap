// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Positive Test - Approval Organizer @VN-4994 @agrannis7311: As a admin, I can approval organizer) 1`] = `
Object {
  "data": Array [
    Object {
      "avatar": "https://reqres.in/img/faces/7-image.jpg",
      "email": "michael.lawson@reqres.in",
      "first_name": "Michael",
      "id": 7,
      "last_name": "Lawson",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/8-image.jpg",
      "email": "lindsay.ferguson@reqres.in",
      "first_name": "Lindsay",
      "id": 8,
      "last_name": "Ferguson",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/9-image.jpg",
      "email": "tobias.funke@reqres.in",
      "first_name": "Tobias",
      "id": 9,
      "last_name": "Funke",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/10-image.jpg",
      "email": "byron.fields@reqres.in",
      "first_name": "Byron",
      "id": 10,
      "last_name": "Fields",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/11-image.jpg",
      "email": "george.edwards@reqres.in",
      "first_name": "George",
      "id": 11,
      "last_name": "Edwards",
    },
    Object {
      "avatar": "https://reqres.in/img/faces/12-image.jpg",
      "email": "rachel.howell@reqres.in",
      "first_name": "Rachel",
      "id": 12,
      "last_name": "Howell",
    },
  ],
  "page": 2,
  "per_page": 6,
  "support": Object {
    "text": "To keep ReqRes free, contributions towards server costs are appreciated!",
    "url": "https://reqres.in/#support-heading",
  },
  "total": 12,
  "total_pages": 2,
}
`;
